src/Mod/Sketcher/Gui/EditDatumDialog.cpp

```diff
@@ -318,4 +326,100 @@ void EditDatumDialog::formEditorOpened(bool state)
     }
 }
 
+
+// This function checks an object's visible flag recursively in a Gui::Document
+// assuming that lastParent (if provided) is visible
+bool isVisibleUpTo(App::DocumentObject* obj, Gui::Document* doc, App::DocumentObject* lastParent)
+{
+    while (obj && obj != lastParent) {
+        auto parentviewprovider = doc->getViewProvider(obj);
+
+        if (!parentviewprovider || !parentviewprovider->isVisible()) {
+            return false;
+        }
+        obj = obj->getFirstParent();
+    }
+    return true;
+}
+bool hasVisualFeature(App::DocumentObject* obj, App::DocumentObject* rootObj, Gui::Document* doc)
+{
+    auto docObjects = doc->getDocument()->getObjects();
+    for (auto object : docObjects) {
+
+        // Presumably, the sketch that is being edited has visual features, but
+        // that's not interesting
+        if (object == obj) {
+            continue;
+        }
+
+        // No need to continue analysis if the object's visible flag is down
+        bool visible = isVisibleUpTo(object, doc, rootObj);
+        if (!visible) {
+            continue;
+        }
+
+        App::DocumentObject* link = object->getLinkedObject();
+        if (link->getDocument() != doc->getDocument()) {
+            Gui::Document* linkDoc = Gui::Application::Instance->getDocument(link->getDocument());
+            if (linkDoc && hasVisualFeature(link, link, linkDoc)) {
+                return true;
+            }
+            continue;
+        }
+
+        // Skip objects that are not of geometric nature
+        if (!object->isDerivedFrom<App::GeoFeature>()) {
+            continue;
+        }
+
+        // Skip datum objects
+        if (object->isDerivedFrom<App::DatumElement>()) {
+            continue;
+        }
+
+        // Skip container objects because getting their bounging box might
+        // return a valid bounding box around annotations or datums
+        if (object->hasExtension(App::GeoFeatureGroupExtension::getExtensionClassTypeId())) {
+            continue;
+        }
+
+        // Get the bounding box of the object
+        auto viewProvider = doc->getViewProvider(object);
+        if (viewProvider && viewProvider->getBoundingBox().IsValid()) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void EditDatumDialog::performAutoScale(double newDatum)
+{
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath(
+        "User parameter:BaseApp/Preferences/Mod/Sketcher/dimensioning");
+    long autoScaleMode = hGrp->GetInt("AutoScaleMode", 0);
+
+    // There is a single constraint in the sketch so it can
+    // be used as a reference to scale the geometries around the origin
+    // if there are external geometries, it is safe to assume that the sketch
+    // was drawn with these geometries as scale references (use <= 2 because
+    // the sketch axis are considered as external geometries)
+    if ((autoScaleMode == 0
```

Comment:
`autoScaleMode` should be an enum probably. It is really hard to understand what `0` or `2` means. Take a look on https://github.com/FreeCAD/FreeCAD/pull/21664 on how it was implemented here.